File: crates/app/src/presentation/item_operations.rs
use crate::i18n::TranslationSet;

pub fn item_operation_to_string(
    item_operation: super_lazygit_tui::presentation::ItemOperation,
    tr: &TranslationSet,
) -> String {
    match item_operation {
        super_lazygit_tui::presentation::ItemOperation::None => String::new(),
        super_lazygit_tui::presentation::ItemOperation::Pushing => tr.pushing_status.clone(),
        super_lazygit_tui::presentation::ItemOperation::Pulling => tr.pulling_status.clone(),
        super_lazygit_tui::presentation::ItemOperation::FastForwarding => tr.fast_forwarding.clone(),
        super_lazygit_tui::presentation::ItemOperation::Deleting => tr.deleting_status.clone(),
        super_lazygit_tui::presentation::ItemOperation::Fetching => tr.fetching_status.clone(),
        super_lazygit_tui::presentation::ItemOperation::CheckingOut => tr.checking_out_status.clone(),
    }
}

File: pkg/gui/presentation/item_operations.go
package presentation

import (
	"github.com/quangdang46/slg/pkg/gui/types"
	"github.com/quangdang46/slg/pkg/i18n"
)

func ItemOperationToString(itemOperation types.ItemOperation, tr *i18n.TranslationSet) string {
	switch itemOperation {
	case types.ItemOperationNone:
		return ""
	case types.ItemOperationPushing:
		return tr.PushingStatus
	case types.ItemOperationPulling:
		return tr.PullingStatus
	case types.ItemOperationFastForwarding:
		return tr.FastForwarding
	case types.ItemOperationDeleting:
		return tr.DeletingStatus
	case types.ItemOperationFetching:
		return tr.FetchingStatus
	case types.ItemOperationCheckingOut:
		return tr.CheckingOutStatus
	}

	return ""
}
